### ISSUE TITLE
No root page found if server runs on specific port

### ISSUE THREAD

**Issue by bezin** [contao/contao#3783]
**Affected version(s)**

Contao 4.9

**Description**

I have a Contao installation that runs on port 8080 in my local dev environment. I used to configure the DNS with port number. 
Since https://github.com/contao/contao/pull/2362 the port number is stripped from the dns config, which breaks routing alltogether. I just experienced this now as it is the first time since a long time I saved the root page, actually running the new checkDns method. So all time until then, the DNS setting was still stored in the database including the port number. 

In short: Running your website under `example.local:8080` with DNS setting `example.local` does not work in Contao 4.9. 

I manually applied the changes made in https://github.com/contao/contao/pull/2673 to `Environment` and `UrlMatcher` and it fixes these issues, too.

**Comment by aschempp** [comment]
Symfony routing does not support port numbers. It simply ignores them. If you access a site with port 8080 and the domain matches, it will happily reply with port 8080 in a URL to that site. There is no need to configure the port. The only thing that is **not** possible is run the same installation with different domains on multiple ports, but there's really no use case for this.

**Comment by bezin** [comment]
I understand that, but [this line ](https://github.com/contao/contao/blob/4.9/core-bundle/src/Routing/Matcher/UrlMatcher.php#L38) sets the host name WITH port on the request context and hence the URL matching fails. It has been removed for 4.10 onwards and it should be removed for 4.9, too.
